app/kuma-dp/cmd/run.go

```diff
@@ -136,15 +136,27 @@ func newRunCmd(opts kuma_cmd.RunCmdOpts, rootCtx *RootContext) *cobra.Command {
 				cfg.Dataplane.Name = proxyResource.GetMeta().GetName()
 			}
 
-			if cfg.DataplaneRuntime.ConfigDir == "" || cfg.DNS.ConfigDir == "" {
+			if cfg.DataplaneRuntime.WorkDir == "" {
+				if cfg.DataplaneRuntime.ConfigDir != "" {
+					runLog.Info("configDir(--config-dir) in config file is deprecated, use workDir(--work-dir) instead")
+					cfg.DataplaneRuntime.WorkDir = cfg.DataplaneRuntime.ConfigDir
+				}
+				if cfg.DataplaneRuntime.SocketDir != "" {
+					runLog.Info("socketDir in config file is deprecated, use workDir(--work-dir) instead")
+				}
+			} else {
+				cfg.DataplaneRuntime.SocketDir = cfg.DataplaneRuntime.WorkDir
+			}
+
+			if cfg.DataplaneRuntime.WorkDir == "" || cfg.DNS.ConfigDir == "" {
```

Comment:
Following the earlier comment
```suggestion
			if cfg.DataplaneRuntime.WorkDir == "" || cfg.DataplaneRuntime.SocketDir == "" || cfg.DNS.ConfigDir == "" {
```